[blog] Announce JS SDK 2.0 (#6574)
open-telemetry/opentelemetry.io: content/en/blog/2025/otel-js-sdk-2-0.md

---
title: Announcing the OpenTelemetry JavaScript SDK 2.0
linkTitle: OTel JS SDK 2.0

> date: 2025-02-21

Reviewer: Setting draft status so that we can see the preview of this post (might have to change the date too, I'm not sure):

```suggestion
date: 2025-02-21
draft: true
```

author: >
  [Jamie Danielson](https://github.com/JamieDanielson) (Honeycomb)
sig: OpenTelemetry JS
cSpell:ignore: Danielson
---

Exciting news!
[OpenTelemetry JavaScript](https://github.com/open-telemetry/opentelemetry-js)
has released
[SDK 2.0](https://github.com/open-telemetry/opentelemetry-js/releases)!

> There is a
> [migration guide](https://github.com/open-telemetry/opentelemetry-js/blob/main/doc/upgrade-to-2.x.md)
> in the repository detailing breaking changes and steps for users to take to
> upgrade to the latest versions.

Reviewer: This paragraph seems redundant given the content below. If you intended this to be a TL;DR or call out, then wrap it in an alert like this:

```suggestion
{{% alert-md title="Migration guide" color="info" %}}

For a detailed description of breaking changes, see the **migration guide**
[Upgrade to OpenTelemetry JS SDK 2.x][migration guide].

[migration guide]:
  https://github.com/open-telemetry/opentelemetry-js/blob/main/doc/upgrade-to-2.x.md

{{% /alert-md %}}
```

## What is JS SDK 2.x?

> "JS SDK 2.x" encompasses new releases of the `@opentelemetry/*` JavaScript

Reviewer: Why the quotes? I suggest dropping them:

```suggestion
JS SDK 2.x encompasses new releases of the `@opentelemetry/*` JavaScript
```

packages published from the
[`opentelemetry-js` repository](https://github.com/open-telemetry/opentelemetry-js),
except the API and semantic-conventions packages. The package versions for this
new major will be `>=2.0.0` for the stable and `>=0.200.0` for the unstable
packages. Details on the full list of packages can be found in the

> [migration guide](https://github.com/open-telemetry/opentelemetry-js/blob/main/doc/upgrade-to-2.x.md#what-is-js-sdk-2x).

Reviewer: If you commit the `{{% alert-md ... %}}` suggested previously, you'll be able to reuse the markdown link definition:

```suggestion
[migration guide][].
```

- [Other changes](https://github.com/open-telemetry/opentelemetry-js/blob/main/doc/upgrade-to-2.x.md#-other-changes)

## Why was this done?

Some of the details for why this 2.0 was done can be found
[in this issue](https://github.com/open-telemetry/opentelemetry-js/issues/4083).
We knew we would gain the greatest benefit by allowing breaking changes to
improve things related to:

- Optimization: removing classes and namespaces to allow better minification and
  tree-shaking.
- Better Tooling and Support: dropping old runtimes and tool versions to take
  advantage of enhanced ESM support and simplify documentation.
- Velocity: reducing code complexity and removing deprecated fields for faster
  feature and maintenance work.

This also begins our goal of releasing a new major release every year.

## How can I get involved?

- Try out the
  [v2.0.0](https://github.com/open-telemetry/opentelemetry-js/releases/tag/v2.0.0)
  and
  [v0.200.0](https://github.com/open-telemetry/opentelemetry-js/releases/tag/experimental%2Fv0.200.0)

>   releases and provide feedback

Reviewer 2: do you want to add something on how to provide feedback? I see people getting confused on how to do so. Maybe mention creation of issues, or reaching our slack/SIG. I know you added slack and sig right below, but wanted to make more clear those are the communications channels for feedback too

---
title: Announcing the OpenTelemetry JavaScript SDK 2.0
linkTitle: OTel JS SDK 2.0

> date: 2025-02-21

Reviewer: ```suggestion
date: 2025-02-20
```

> date: 2025-02-21
> draft: true

Reviewer: Oops, looked too quickly, this date should be in March not February :). Fixing that:

```suggestion
date: 2025-03-21
```

We shouldn't need the draft status.

---
title: Announcing the OpenTelemetry JavaScript SDK 2.0
linkTitle: OTel JS SDK 2.0
date: 2025-03-21
author: >
  [Jamie Danielson](https://github.com/JamieDanielson) (Honeycomb)
sig: OpenTelemetry JS
cSpell:ignore: Danielson
---

> Exciting news: [OpenTelemetry JavaScript](/docs/languages/js/) has released
> [SDK 2.0](https://github.com/open-telemetry/opentelemetry-js/releases/tag/v2.0.0)!

Reviewer: ```suggestion
Exciting news: [OpenTelemetry JavaScript](/docs/languages/js/) has released [SDK
2.0][v2.0.0]!

[v2.0.0]: https://github.com/open-telemetry/opentelemetry-js/releases/tag/v2.0.0
```

FYI: the same link definition is used further below in another suggestion.

{{% alert-md title="Migration guide" color="info" %}}

For a detailed description of breaking changes, see the **migration guide**
[Upgrade to OpenTelemetry JS SDK 2.x][migration guide].

[migration guide]:
  https://github.com/open-telemetry/opentelemetry-js/blob/main/doc/upgrade-to-2.x.md

{{% /alert-md %}}

## What is JS SDK 2.x?

JS SDK 2.x encompasses new releases of the `@opentelemetry/*` JavaScript
packages published from the
[`opentelemetry-js` repository](https://github.com/open-telemetry/opentelemetry-js),
except the API and semantic-conventions packages. The package versions for this
new major will be `>=2.0.0` for the stable and `>=0.200.0` for the unstable
packages. Details on the full list of packages can be found in the [migration
guide][].

## What has changed?

> ### Summary

Reviewer: ```suggestion
In summary:
```

- The **minimum supported Node.js has been raised to `^18.19.0 || >=20.6.0`**.
  This means that support for Node.js 14 and 16 has been dropped.
- The **minimum supported TypeScript version has been raised to 5.0.4**.
- The **compilation target for transpiled TypeScript has been raised to ES2022**
  (from ES2017).
- The **public interface has changed**
  - for notes on migrating to 2.x / 0.200.x see
    [the upgrade guide](https://github.com/open-telemetry/opentelemetry-js/tree/main/doc/upgrade-to-2.x.md)

> ### Details

Reviewer: ```suggestion
Details:
```

- [Drop `window.OTEL_*` support in browsers](https://github.com/open-telemetry/opentelemetry-js/blob/main/doc/upgrade-to-2.x.md#-drop-windowotel_-support-in-browsers)
- [`@opentelemetry/resources` API changes](https://github.com/open-telemetry/opentelemetry-js/blob/main/doc/upgrade-to-2.x.md#-opentelemetryresources-api-changes)
- [`@opentelemetry/core` API changes](https://github.com/open-telemetry/opentelemetry-js/blob/main/doc/upgrade-to-2.x.md#-opentelemetrycore-api-changes)
- [Tracing SDK API changes](https://github.com/open-telemetry/opentelemetry-js/blob/main/doc/upgrade-to-2.x.md#-tracing-sdk-api-changes)
- [`@opentelemetry/sdk-metrics` API changes](https://github.com/open-telemetry/opentelemetry-js/blob/main/doc/upgrade-to-2.x.md#-opentelemetrysdk-metrics-api-changes)
- [`@opentelemetry/resources` changes for _implementors_ of Resource Detectors](https://github.com/open-telemetry/opentelemetry-js/blob/main/doc/upgrade-to-2.x.md#-opentelemetryresources-changes-for-implementors-of-resource-detectors)
- [Other changes](https://github.com/open-telemetry/opentelemetry-js/blob/main/doc/upgrade-to-2.x.md#-other-changes)

## Why was this done?

> Some of the details for why this 2.0 was done can be found
> [in this issue](https://github.com/open-telemetry/opentelemetry-js/issues/4083).
> We knew we would gain the greatest benefit by allowing breaking changes to
> improve things related to:
> 
> - Optimization: removing classes and namespaces to allow better minification and
>   tree-shaking.
> - Better Tooling and Support: dropping old runtimes and tool versions to take
>   advantage of enhanced ESM support and simplify documentation.
> - Velocity: reducing code complexity and removing deprecated fields for faster
>   feature and maintenance work.
> 
> This also begins our goal of releasing a new major release every year.

Reviewer: Move the "for more details" to the end of the post:

```suggestion
We knew we would gain the greatest benefit by allowing breaking changes to
improve things related to:

- Optimization: removing classes and namespaces to allow better minification and
  tree-shaking.
- Better Tooling and Support: dropping old runtimes and tool versions to take
  advantage of enhanced ESM support and simplify documentation.
- Velocity: reducing code complexity and removing deprecated fields for faster
  feature and maintenance work.

This also begins our goal of releasing a new major release every year.

For a more detailed explanation of why 2.0, see issue [#4083].

[#4083]: https://github.com/open-telemetry/opentelemetry-js/issues/4083
```

> This also begins our goal of releasing a new major release every year.

Reviewer 3: Should we emphasize this applies to "SDK"?

```suggestion
This also begins our goal of releasing a new SDK major release every year.
```

## How can I get involved?

> - Try out the
>   [v2.0.0](https://github.com/open-telemetry/opentelemetry-js/releases/tag/v2.0.0)
>   and
>   [v0.200.0](https://github.com/open-telemetry/opentelemetry-js/releases/tag/experimental%2Fv0.200.0)
>   releases and provide feedback

Reviewer: ```suggestion
[v0.200.0]:
  https://github.com/open-telemetry/opentelemetry-js/releases/tag/experimental%2Fv0.200.0

- Try out the [v2.0.0] and [v0.200.0] releases and provide feedback
```

---
title: Announcing the OpenTelemetry JavaScript SDK 2.0
linkTitle: OTel JS SDK 2.0

> date: 2025-03-21

Author: ```suggestion
date: 2025-03-26
```

[v2.0.0]: https://github.com/open-telemetry/opentelemetry-js/releases/tag/v2.0.0

{{% alert-md title="Migration guide" color="info" %}}

For a detailed description of breaking changes, see the **migration guide**
[Upgrade to OpenTelemetry JS SDK 2.x][migration guide].

[migration guide]:
  https://github.com/open-telemetry/opentelemetry-js/blob/main/doc/upgrade-to-2.x.md

{{% /alert-md %}}

## What is JS SDK 2.x?

JS SDK 2.x encompasses new releases of the `@opentelemetry/*` JavaScript
packages published from the
[`opentelemetry-js` repository](https://github.com/open-telemetry/opentelemetry-js),
except the API and semantic-conventions packages. The package versions for this
new major will be `>=2.0.0` for the stable and `>=0.200.0` for the unstable
packages. Details on the full list of packages can be found in the [migration
guide][].

## What has changed?

In summary:

- The **minimum supported Node.js has been raised to `^18.19.0 || >=20.6.0`**.
  This means that support for Node.js 14 and 16 has been dropped.
- The **minimum supported TypeScript version has been raised to 5.0.4**.
- The **compilation target for transpiled TypeScript has been raised to ES2022**
  (from ES2017).

> - The **public interface has changed**
>   - for notes on migrating to 2.x / 0.200.x see
>     [the upgrade guide](https://github.com/open-telemetry/opentelemetry-js/tree/main/doc/upgrade-to-2.x.md)

Reviewer 4: ```suggestion
- The **public interface has changed**.
  - For notes on migrating to 2.x / 0.200.x, see
    [the upgrade guide](https://github.com/open-telemetry/opentelemetry-js/tree/main/doc/upgrade-to-2.x.md).
```

- [`@opentelemetry/resources` API changes](https://github.com/open-telemetry/opentelemetry-js/blob/main/doc/upgrade-to-2.x.md#-opentelemetryresources-api-changes)
- [`@opentelemetry/core` API changes](https://github.com/open-telemetry/opentelemetry-js/blob/main/doc/upgrade-to-2.x.md#-opentelemetrycore-api-changes)
- [Tracing SDK API changes](https://github.com/open-telemetry/opentelemetry-js/blob/main/doc/upgrade-to-2.x.md#-tracing-sdk-api-changes)
- [`@opentelemetry/sdk-metrics` API changes](https://github.com/open-telemetry/opentelemetry-js/blob/main/doc/upgrade-to-2.x.md#-opentelemetrysdk-metrics-api-changes)
- [`@opentelemetry/resources` changes for _implementors_ of Resource Detectors](https://github.com/open-telemetry/opentelemetry-js/blob/main/doc/upgrade-to-2.x.md#-opentelemetryresources-changes-for-implementors-of-resource-detectors)
- [Other changes](https://github.com/open-telemetry/opentelemetry-js/blob/main/doc/upgrade-to-2.x.md#-other-changes)

## Why was this done?

We knew we would gain the greatest benefit by allowing breaking changes to
improve things related to:

- Optimization: removing classes and namespaces to allow better minification and
  tree-shaking.

> - Better Tooling and Support: dropping old runtimes and tool versions to take

Reviewer 4: ```suggestion
- Better tooling and support: dropping old runtimes and tool versions to take
```

  advantage of enhanced ESM support and simplify documentation.
- Velocity: reducing code complexity and removing deprecated fields for faster
  feature and maintenance work.

This also begins our goal of releasing a new SDK major release every year.

For a more detailed explanation of why 2.0, see issue [#4083].

[#4083]: https://github.com/open-telemetry/opentelemetry-js/issues/4083

## How can I get involved?

[v0.200.0]:
  https://github.com/open-telemetry/opentelemetry-js/releases/tag/experimental%2Fv0.200.0

> - Try out the [v2.0.0] and [v0.200.0] releases and provide feedback
> - Review our
>   [contributing guide](https://github.com/open-telemetry/opentelemetry-js/blob/main/CONTRIBUTING.md)
>   for details on contributing
> - Attend our
>   [SIG meetings](https://groups.google.com/a/opentelemetry.io/g/calendar-js)
> - Collaborate on [Slack](https://cloud-native.slack.com/archives/C01NL1GRPQR)

Reviewer 4: ```suggestion
- Try out the [v2.0.0] and [v0.200.0] releases and provide feedback.
- Review our
  [contributing guide](https://github.com/open-telemetry/opentelemetry-js/blob/main/CONTRIBUTING.md)
  for details on contributing.
- Attend our
  [SIG meetings](https://groups.google.com/a/opentelemetry.io/g/calendar-js).
- Collaborate on [Slack](https://cloud-native.slack.com/archives/C01NL1GRPQR).
```
